src/api/spec/models/attrib_spec.rb

```diff
@@ -6,14 +6,44 @@
   let(:package) { create(:package) }
 
   context "#container=" do
-    it "sets the project" do
-      attribute.container = project
-      expect(attribute.container).to be(project)
+    context "assigning a project" do
+      before do
+        attribute.container = project
+      end
+
+      it { expect(attribute.container).to be(project) }
+      it { expect(attribute.project).to be(project) }
+      it { expect(attribute.package).to be_nil }
+
+      context "and then assigning a package" do
+        before do
+          attribute.container = package
+        end
+
+        it { expect(attribute.container).to be(package) }
+        it { expect(attribute.project).to be(package.project) }
+        it { expect(attribute.package).to be(package) }
+      end
     end
 
-    it "sets the package" do
-      attribute.container = package
-      expect(attribute.container).to be(package)
+    context "assigning a package" do
```

Comment:
Isn't that exactly the same test as line 9-25 just the other way around? Is that really necessary?